### ISSUE TITLE
ipw计算中的问题

### ISSUE THREAD

**Issue by c00h00g** [orangeai-research/ESCM2#2]
ctr_num = ctr_clk.sum()  
        PS = ctr_out_one * ctr_num.float()  
        PS = torch.clamp(PS, min=1e-6)  # Avoid division by zero  
        IPS = 1.0 / PS  
        IPS = torch.clamp(IPS, min=-15, max=15)  
        #IPS.requires_grad_(False)  # Stop gradient  
        IPS = IPS.detach()
        batch_size = float(ctr_clk.size(0))  
        IPS_scaled = IPS * batch_size  
        loss_cvr_weighted = loss_cvr * IPS_scaled  
        loss_cvr_final = (loss_cvr_weighted * ctr_clk_float).mean()  
        return loss_cvr_final  

请教下，
1. 我看了下由于loss_cvr = (1/batch)*sum(loss),  因此才做了batch/ctr_num的转换，为啥不直接按照公式计算呢？比如先计算出来ips([batch, 1]), 再计算出来loss([batch,1])，然后相乘呢？

2. 公式里的分母是D，现在看起来是ctr_num了，是不是和论文不一致？
![image](https://github.com/user-attachments/assets/fdafb474-97d5-4657-9042-6cd51e4954a8)
